addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -1747,24 +1803,31 @@ export function rgbToHex(rgb = '') {
     );
 }
 
-export const leftDeepFirstPath = createDOMPathGenerator(DIRECTIONS.LEFT, false, false);
-export const leftDeepOnlyPath = createDOMPathGenerator(DIRECTIONS.LEFT, true, false);
-export const leftDeepFirstInlinePath = createDOMPathGenerator(DIRECTIONS.LEFT, false, true);
-export const leftDeepOnlyInlinePath = createDOMPathGenerator(DIRECTIONS.LEFT, true, true);
-export const leftDeepOnlyInlineInScopePath = createDOMPathGenerator(
-    DIRECTIONS.LEFT,
-    true,
-    true,
-    true,
-);
-
-export const rightDeepFirstPath = createDOMPathGenerator(DIRECTIONS.RIGHT, false, false);
-export const rightDeepOnlyPath = createDOMPathGenerator(DIRECTIONS.RIGHT, true, false);
-export const rightDeepFirstInlinePath = createDOMPathGenerator(DIRECTIONS.RIGHT, false, true);
-export const rightDeepOnlyInlinePath = createDOMPathGenerator(DIRECTIONS.RIGHT, true, true);
-export const rightDeepOnlyInlineInScopePath = createDOMPathGenerator(
-    DIRECTIONS.RIGHT,
-    true,
-    true,
-    true,
-);
+export const leftLeafFirstPath = createDOMPathGenerator(DIRECTIONS.LEFT);
+export const leftLeafOnlyNotBlockPath = createDOMPathGenerator(DIRECTIONS.LEFT, {
+    leafOnly: true,
+    isNodeDescendantTraversable: isBlock,
+    stopFunction: isBlock,
+});
+export const leftLeafOnlyInScopeNotBlockPath = createDOMPathGenerator(DIRECTIONS.LEFT, {
```

Comment:
![image](https://user-images.githubusercontent.com/24205914/126989895-935e9c9b-67d4-4a25-8b93-cdabd0dd73d5.png)

Reply:
Anywhere we use the function we would need to name them.
I agree that the name are confusing but it would not change if we define the function somewhere else.
At least now, we can see them all at once which make their edition and comparing between them easier.
So I would be against moving them even if used only once.

Reply:
I asked because you said reading the arguments was the easiest way to understand. If that's true, we should be able to just call them `path` wherever we need them and the list of arguments would speak of itself.
Are they each used several times?
I went to check and actually they're basically all either used once or never. Only `leftDeepOnlyInlinePath` and `rightDeepOnlyInlinePath` are used twice (in `getState` and in `enforceWhitespace`).
Now I do hear your argument so if you want to leave it as is I won't block the PR for that but I thought it would be worth considering getting rid of this altogether (there are pros and cons).